autoload/neomake.vim

```diff
@@ -484,106 +501,124 @@ function! s:HandleLoclistQflistDisplay(file_mode) abort
     endif
 endfunction
 
-function! s:Make(options, ...) abort
-    let file_mode = get(a:options, 'file_mode', 0)
-    let enabled_makers = get(a:options, 'enabled_makers', [])
-    if !len(enabled_makers)
-        if file_mode
-            call neomake#utils#DebugMessage('Nothing to make: no enabled makers.')
-            return []
+function! s:Make(options) abort
+    let options = copy(a:options)
+    call extend(options, {
+                \ 'file_mode': 0,
+                \ 'enabled_makers': [],
+                \ 'bufnr': bufnr('%'),
+                \ 'ft': '',
+                \ }, 'keep')
+    let bufnr = options.bufnr
+    let file_mode = options.file_mode
+    let enabled_makers = options.enabled_makers
+    let ft = options.ft
+
+    " Reset/clear on first run, but not when using 'serialize'.
+    if !get(options, 'continuation', 0)
+        if !len(enabled_makers)
+            if file_mode
+                call neomake#utils#DebugMessage('Nothing to make: no enabled makers.')
+                return []
+            endif
+            let enabled_makers = ['makeprg']
         endif
-        let enabled_makers = ['makeprg']
-    endif
 
-    if a:0
-        let make_id = a:1
-    else
         let s:make_id += 1
-        let make_id = s:make_id
-    endif
-
-    let buf = bufnr('%')
-    let ft = get(a:options, 'ft', '')
-
-    if file_mode
-        call neomake#statusline#ResetCountsForBuf(buf)
-    else
-        call neomake#statusline#ResetCountsForProject()
-    endif
+        let s:make_options[s:make_id] = {
+                    \ 'cwd': getcwd(),
+                    \ }
```

Comment:
Move this to `maker` again?!
This would allow to dynamically change it etc, and remove the new dict.